crates/shared/src/oneinch_api.rs

```diff
@@ -496,7 +503,12 @@ impl OneInchClientImpl {
     pub const SUPPORTED_CHAINS: &'static [u64] = &[1, 100];
 
     /// Create a new 1Inch HTTP API client with the specified base URL.
-    pub fn new(base_url: impl IntoUrl, client: Client, chain_id: u64) -> Result<Self> {
+    pub fn new(
+        base_url: impl IntoUrl,
+        client: Client,
```

Comment:
Did you consider wrapping `reqwest::Client` into a services-specific HTTP client that:
1. Does `trace` logging (instead of duplicating it everywhere)
2. Adds all the required headers that we care about (like the block hash header)?

This way, we don't have to pass two things around everywhere and abstract it all away as the HTTP client we use.